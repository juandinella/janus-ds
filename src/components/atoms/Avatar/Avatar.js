import isPropValid from '@emotion/is-prop-valid'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import styles from './Avatar.styles'
import { options } from './constants'

/**
 * Avatar
 */

const shouldForwardProp = (prop) => isPropValid(prop)
const StyledAvatar = styled.div.withConfig({shouldForwardProp})`${styles}`

const Avatar = ({ id, size, isOnline, children }) => {
  return <StyledAvatar id={id} size={size} isOnline={isOnline}>{children}</StyledAvatar>
}

Avatar.displayName = 'Avatar'

Avatar.propTypes = {
  /** Component id */
  id: PropTypes.string,
  /** Content to be included in the component */
  children: PropTypes.node.isRequired,
  /** Size of the component */
  size: PropTypes.oneOf(options.sizes),
  /** It shows online status */
  isOnline: PropTypes.bool,
}

Avatar.defaultProps = {}

export default Avatar
