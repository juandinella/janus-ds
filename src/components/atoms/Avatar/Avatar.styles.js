import { css } from 'styled-components'

const avatarSizes = {
  sm: css`
    --size: 28px;
    font-size: var(--font-size-xs);
  `,
  md: css`
    --size: 38px;
    font-size: var(--font-size-md);
  `,
  lg: css`
    --size: 42px;
    font-size: var(--font-size-lg);
  `,
}

export default css`
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: var(--bg-primary-lighter);
  color: var(--text-color-primary-darker);
  font-weight: var(--font-weight-semi);
  border-radius: var(--border-radius-md);
  text-transform: uppercase;
  height: var(--size);
  width: var(--size);

  ${(props) => props.size && avatarSizes[props.size]}

  ${(props) =>
    props.isOnline &&
    css`
      position: relative;

      &:after {
        content: '';
        width: 14px;
        height: 14px;
        background-color: var(--bg-success);
        position: absolute;
        bottom: -2px;
        right: -4px;
        border-radius: 100%;
        border: 2px solid white;
      }
    `}
`
