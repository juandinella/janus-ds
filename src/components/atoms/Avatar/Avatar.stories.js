import Avatar from './Avatar'
import { options } from './constants'

import {
  getTemplate,
  getListTemplate,
  getOptionsArgTypes,
} from '../../../helpers/storybook'

const Template = getTemplate(Avatar)
const ListTemplate = getListTemplate(Avatar)

export default {
  title: 'Atoms/Avatar',
  component: Avatar,
  args: {
    children: 'gr',
    size: 'md',
  },
  argTypes: {
    ...getOptionsArgTypes(options.sizes, 'size'),
  },
}

export const Default = Template.bind({})

export const Online = { args: { size: 'lg', isOnline: true } }

export const Sizes = ListTemplate.bind({})
Sizes.args = { items: options.sizes.map((size) => ({ size })) }
