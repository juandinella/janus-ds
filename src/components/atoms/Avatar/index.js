import Avatar from './Avatar'

export default Avatar
export { options } from './constants'
