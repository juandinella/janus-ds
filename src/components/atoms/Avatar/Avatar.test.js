import initStoryshots from '@storybook/addon-storyshots'
import { render, screen, cleanup } from '@testing-library/react'
import React from 'react'
import Avatar from './Avatar'

initStoryshots({
  storyKindRegex: /^Atoms\/Avatar$/,
})

const componentProps = {
  id: 'unique_id',
  children: 'Soy un mensaje de prueba',
  size: 'md',
  isOnline: true,
}
afterEach(cleanup)

describe('<Avatar />', () => {
  test('should render component correctly', () => {
    render(<Avatar {...componentProps} />)
    const avatar = screen.getByTestId(componentProps.id)
    expect(avatar).toBeInTheDocument()
  })
  test('should render children correctly', () => {
    render(<Avatar {...componentProps} />)
    const childrenElement = screen.getByTestId(`${componentProps.id}-children`)
    expect(childrenElement.textContent).toEqual(componentProps.children)
  })
})
