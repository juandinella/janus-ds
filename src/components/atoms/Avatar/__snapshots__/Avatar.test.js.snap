// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Avatar Default 1`] = `
<div
  style={
    {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "position": "relative",
    }
  }
>
  <div
    className="sc-odj6ky-0 hcSgNG"
    size="md"
  >
    <span
      data-testid="undefined-children"
    >
      gr
    </span>
  </div>
</div>
`;

exports[`Storyshots Atoms/Avatar Online 1`] = `
<div
  style={
    {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "position": "relative",
    }
  }
>
  <div
    className="sc-odj6ky-0 bULrQs"
    size="lg"
  >
    <span
      data-testid="undefined-children"
    >
      gr
    </span>
  </div>
</div>
`;

exports[`Storyshots Atoms/Avatar Sizes 1`] = `
<div
  style={
    {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "position": "relative",
    }
  }
>
  <div
    className="sc-odj6ky-0 lojJIv"
    size="sm"
  >
    <span
      data-testid="undefined-children"
    >
      gr
    </span>
  </div>
  <div
    className="sc-odj6ky-0 hcSgNG"
    size="md"
  >
    <span
      data-testid="undefined-children"
    >
      gr
    </span>
  </div>
  <div
    className="sc-odj6ky-0 cUplmE"
    size="lg"
  >
    <span
      data-testid="undefined-children"
    >
      gr
    </span>
  </div>
</div>
`;
